velox/common/caching/StringIdMap.cpp

```diff
@@ -74,7 +73,8 @@ uint64_t StringIdMap::makeId(std::string_view string) {
   // be in the 100K range.
   do {
     entry.id = ++lastId_;
-  } while (idToString_.find(entry.id) != idToString_.end());
+  } while (entry.id == kNoId ||
```

Comment:
There is no guarantee that `kNoId + 1` is unused.  You need to track the oldest used ID if you really want to make it working correctly in case of overflow.

Reply:
Hi @Yuhta , I have added UT, please take a look again. The reason I consider overflow is because I think the do-while block itself is to make it work when overflow happens, as the comments indicates
```
  // Check that we do not use an id twice. In practice this never
  // happens because the int64 counter would have to wrap around for
  // this. 
```

Reply:
BTW, the CI error seems not related to this PR.